packages/client/wallets/smart-wallet/src/blockchain/wallets/SendTransactionService.ts

```diff
@@ -0,0 +1,142 @@
+import { SmartAccountClient } from "permissionless";
+import { EntryPoint } from "permissionless/_types/types";
+import { SmartAccount } from "permissionless/accounts";
+import {
+    Abi,
+    Address,
+    BaseError,
+    Chain,
+    ContractFunctionRevertedError,
+    PublicClient,
+    TransactionReceipt,
+    Transport,
+} from "viem";
+
+import { CrossmintSDKError, WalletErrorCode } from "@crossmint/client-sdk-base";
+
+export type TransactionServiceTransactionRequest = {
+    address: Address;
+    abi: Abi;
+    functionName: string;
+    args: any;
+    value?: bigint;
+};
+
+/**
+ * Error thrown when a transaction fails to send.
+ * @param viemError The error thrown by the viem client.
+ */
+export class SendTransactionError extends CrossmintSDKError {
```

Comment:
The wallets SDK is going to support solana and other chains in the future, that don't use viem

As these errors are coupled to viem (whcih is fine), should we maybe include EVM somewhere in the error class name? 

e.g. EVMSendTransactionError

Reply:
Changed